lib/compiler/ast-to-ops.ts

```diff
@@ -730,8 +737,74 @@ function compileStatementToOp(statement : Ast.Rule|Ast.Command) : RuleOp {
     return new RuleOp(streamop, statement.actions, statement);
 }
 
+function compileBooleanExpressionToOp(expr : Ast.BooleanExpression) : BooleanExpressionOp {
+    if (expr instanceof Ast.AtomBooleanExpression)
```

Comment:
If we're going the route of an intermediate representation for boolean expressions, we should probably remove the distinction of AtomBooleanExpression and ComputeBooleanExpression at the "op" level.

Reply:
why do we want to do that? these two ops are compiled differently, we will need to distinguish them by fields anyway if we merge them.

Reply:
Because you don't need to compile them differently: an AtomBooleanExpression is the same as a ComputeBooleanExpression where the lhs is a VarRefValue.